### PR TITLE
DEV: Allows server_plugin_outlet to use local variable

### PR DESCRIPTION
**Pull request by Ahmedgagan**
@jjaffeux

### CODE REVIEW

**Comment by SamSaffron** [comment]
@jjaffeux what is the context here?

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -390,7 +389,19 @@ def server_plugin_outlet(name)
     return "" if erbs.blank?
 
     result = +""
-    erbs.each { |erb| result << render(inline: File.read(erb)) }
+    unless var
```
Comment (SamSaffron) [inline app/helpers/application_helper.rb:392]: not sure if we are going to merge this, but we certainly should never ever do `unless ... else` ... `if ... else` is far less confusing.

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -390,7 +389,19 @@ def server_plugin_outlet(name)
...
+      erbs.each { |erb| result << render(inline: File.read(erb)) }
+    else
+      erbs.each do |erb|
+        file_name = erb.split("/").last
+
+        file_name = file_name.slice((file_name.index("_") + 1)..-1)
+        paths = erb.split("views")
+
+        result << render("/connectors/#{name}/#{file_name}", object: var)
```
Comment (SamSaffron) [inline app/helpers/application_helper.rb:401]: something about the code her fees fragile.

Comment (SamSaffron) [reply app/helpers/application_helper.rb:401]: also must have a test for this key functionality.

**Comment by Ahmedgagan** [comment]
> @jjaffeux what is the context here?

In server_plugin_outlet we can not pass the data to the view because inline render does not use passed local data.
So, I did these changes to pass the data to the views using server_plugin_outlets.

I needed it to add a field in the embed topic view.

**Comment by SamSaffron** [comment]
I see, I assume we have a workaround for now. Going to close this.

**Pull request closed without merging**